runtime/vm/jfr.cpp

```diff
@@ -513,35 +513,47 @@ jfrVMSlept(J9HookInterface **hook, UDATA eventNum, void *eventData, void *userDa
 /**
  * Hook for VM intialized. Called without VM access.
  *
- * @param hook[in] the VM hook interface
- * @param eventNum[in] the event number
+ * @param hook[in] the VM hook interface, not used
+ * @param eventNum[in] the event number, not used
  * @param eventData[in] the event data
- * @param userData[in] the registered user data
+ * @param userData[in] the registered user data, not used
  */
 static void
 jfrVMInitialized(J9HookInterface **hook, UDATA eventNum, void *eventData, void *userData)
 {
-	J9VMInitEvent *event = (J9VMInitEvent *)eventData;
-	J9VMThread *currentThread = event->vmThread;
-	J9JavaVM *vm = currentThread->javaVM;
+	J9VMThread *currentThread = ((J9VMInitEvent *)eventData)->vmThread;
 
 #if defined(DEBUG)
 	PORT_ACCESS_FROM_VMC(currentThread);
 	j9tty_printf(PORTLIB, "\n!!! VM init %p\n", currentThread);
 #endif /* defined(DEBUG) */
+	jfrStartSamplingThread(currentThread->javaVM);
+}
 
-	/* Start the sampler thread */
-	if (0 == omrthread_create(&(vm->jfrSamplerThread), vm->defaultOSStackSize, J9THREAD_PRIORITY_NORMAL, FALSE, jfrSamplingThreadProc, (void*)vm)) {
+/**
+ * Start JFR sampling thread. Called without VM access.
+ *
+ * @param vm[in] pointer to the J9JavaVM
+ */
+static void
+jfrStartSamplingThread(J9JavaVM *vm)
+{
+	/* Start the sampler thread. */
+	IDATA rc = omrthread_create(&(vm->jfrSamplerThread), vm->defaultOSStackSize, J9THREAD_PRIORITY_NORMAL, FALSE, jfrSamplingThreadProc, (void*)vm);
+	if (0 == rc) {
 		omrthread_monitor_enter(vm->jfrSamplerMutex);
 		while (J9JFR_SAMPLER_STATE_UNINITIALIZED == vm->jfrSamplerState) {
 			omrthread_monitor_wait(vm->jfrSamplerMutex);
 		}
 		omrthread_monitor_exit(vm->jfrSamplerMutex);
-		if (J9JFR_SAMPLER_STATE_DEAD == vm->jfrSamplerState) {
-			// TODO: tracepoint?
+		if (J9JFR_SAMPLER_STATE_RUNNING == vm->jfrSamplerState) {
+			vm->jfrState.isStarted = TRUE;
```

Comment:
This should be set in `initializeJFR` after we know initialization is successful. In the future we will make sampling optional

Reply:
Moved to `initializeJFR()`.